plugins/package-managers/maven/src/main/kotlin/Tycho.kt

```diff
@@ -0,0 +1,236 @@
+        val (_, buildLog) = runBuild(collector, definitionFile.parentFile)
+        // TODO: Create issues for a failed build and projects for which dependencies could not be resolved.
+
+        val resolvedProjects = createMavenSupport(collector).use { mavenSupport ->
+            val dependencyHandler =
+                MavenDependencyHandler(managerName, projectType, mavenSupport, collector.mavenProjects, false)
+
+            graphBuilder = DependencyGraphBuilder<DependencyNode>(dependencyHandler)
+
+            BufferedReader(InputStreamReader(buildLog.inputStream())).use { reader ->
+                parseMavenOutput(reader, collector.mavenProjects.values).map { (project, dependencies) ->
+                    processProjectDependencies(graphBuilder, project, dependencies)
+                    project
+                }
+            }
+        }
+
+        buildLog.delete()
+
+        return resolvedProjects.map { mavenProject ->
+            val projectId = mavenProject.identifier(projectType)
+            val project = mavenProject.toOrtProject(
+                projectId,
+                mavenProject.file,
+                mavenProject.file.parentFile,
+                graphBuilder.scopesFor(projectId)
+            )
+            ProjectAnalyzerResult(project, emptySet())
+        }
+    }
+
+    override fun createPackageManagerResult(projectResults: ProjectResults): PackageManagerResult =
+        PackageManagerResult(projectResults, graphBuilder.build(), graphBuilder.packages())
+
+    /**
+     * Create the [MavenCli] instance to trigger a build on the analyzed project. Register the given [collector], so
+     * that it is invoked during the build.
+     */
+    internal fun createMavenCli(collector: TychoProjectsCollector): MavenCli =
+        object : MavenCli(ClassWorld("plexus.core", javaClass.classLoader)) {
+            override fun customizeContainer(container: PlexusContainer) {
+                container.addComponent(
+                    collector,
+                    AbstractMavenLifecycleParticipant::class.java,
+                    "TychoProjectsCollector"
+                )
+
+                (container as? DefaultPlexusContainer)?.loggerManager = object : BaseLoggerManager() {
+                    override fun createLogger(name: String): org.codehaus.plexus.logging.Logger =
+                        if (DependencyTreeLogger.DEPENDENCY_TREE_LOGGER == name) {
+                            DependencyTreeLogger()
+                        } else {
+                            MavenLogger(logger.delegate.level)
+                        }
+                }
+            }
+        }
+
+    /**
+     * Run a Maven build on the Tycho project in [projectRoot] utilizing the given [collector]. Return a pair with
+     * the exit code of the Maven project and a [File] that contains the output generated during the build.
+     */
+    private fun runBuild(collector: TychoProjectsCollector, projectRoot: File): Pair<Int, File> {
```

Comment:
I haven't check in detail, but my gut feeling is that this functionality should be implemented with the [Maven Invoker API](https://maven.apache.org/shared/maven-invoker/index.html) instead. That's also the approach that I explained to have played with in my [`maven-invoker-api` branch](https://github.com/oss-review-toolkit/ort/tree/maven-invoker-api).

Reply:
Based on a gut feeling, I would rather not do a complete rewrite of my approach.

I also did not check in detail, but I had a short look at the Invoker API. I did not follow this approach because AIUI it requires a Maven installation to be present, which would have quite some impact (e.g. Docker images would need to be extended). So I came to the conclusion that what I am doing now is closer to the existing Maven implementation.

Reply:
> AIUI it requires a Maven installation to be present

You actually seem to be correct here. In that case, using the [Maven Embedder API](https://maven.apache.org/ref/3.8.8/maven-embedder/index.html) should indeed be correct way to go.